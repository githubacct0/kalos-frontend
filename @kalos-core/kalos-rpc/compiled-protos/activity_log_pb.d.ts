// package: 
// file: activity_log.proto

import * as jspb from "google-protobuf";
import * as user_pb from "./user_pb";

export class ActivityLog extends jspb.Message {
  getId(): number;
  setId(value: number): void;

  getUserId(): number;
  setUserId(value: number): void;

  getActivityName(): string;
  setActivityName(value: string): void;

  getActivityDate(): string;
  setActivityDate(value: string): void;

  getPropertyId(): number;
  setPropertyId(value: number): void;

  getContractId(): number;
  setContractId(value: number): void;

  getGeolocationLat(): number;
  setGeolocationLat(value: number): void;

  getGeolocationLng(): number;
  setGeolocationLng(value: number): void;

  getCustomerId(): number;
  setCustomerId(value: number): void;

  getTaskId(): number;
  setTaskId(value: number): void;

  getTimesheetLineId(): number;
  setTimesheetLineId(value: number): void;

  getEventId(): number;
  setEventId(value: number): void;

  clearFieldMaskList(): void;
  getFieldMaskList(): Array<string>;
  setFieldMaskList(value: Array<string>): void;
  addFieldMask(value: string, index?: number): string;

  getPageNumber(): number;
  setPageNumber(value: number): void;

  getOrderBy(): string;
  setOrderBy(value: string): void;

  getOrderDir(): string;
  setOrderDir(value: string): void;

  clearDateRangeList(): void;
  getDateRangeList(): Array<string>;
  setDateRangeList(value: Array<string>): void;
  addDateRange(value: string, index?: number): string;

  getWithUser(): boolean;
  setWithUser(value: boolean): void;

  hasUser(): boolean;
  clearUser(): void;
  getUser(): user_pb.User | undefined;
  setUser(value?: user_pb.User): void;

  getTimeoffRequestId(): number;
  setTimeoffRequestId(value: number): void;

  getDateTarget(): string;
  setDateTarget(value: string): void;

  clearNotEqualsList(): void;
  getNotEqualsList(): Array<string>;
  setNotEqualsList(value: Array<string>): void;
  addNotEquals(value: string, index?: number): string;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): ActivityLog.AsObject;
  static toObject(includeInstance: boolean, msg: ActivityLog): ActivityLog.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: ActivityLog, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): ActivityLog;
  static deserializeBinaryFromReader(message: ActivityLog, reader: jspb.BinaryReader): ActivityLog;
}

export namespace ActivityLog {
  export type AsObject = {
    id: number,
    userId: number,
    activityName: string,
    activityDate: string,
    propertyId: number,
    contractId: number,
    geolocationLat: number,
    geolocationLng: number,
    customerId: number,
    taskId: number,
    timesheetLineId: number,
    eventId: number,
    fieldMaskList: Array<string>,
    pageNumber: number,
    orderBy: string,
    orderDir: string,
    dateRangeList: Array<string>,
    withUser: boolean,
    user?: user_pb.User.AsObject,
    timeoffRequestId: number,
    dateTarget: string,
    notEqualsList: Array<string>,
  }
}

export class ActivityLogList extends jspb.Message {
  clearResultsList(): void;
  getResultsList(): Array<ActivityLog>;
  setResultsList(value: Array<ActivityLog>): void;
  addResults(value?: ActivityLog, index?: number): ActivityLog;

  getTotalCount(): number;
  setTotalCount(value: number): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): ActivityLogList.AsObject;
  static toObject(includeInstance: boolean, msg: ActivityLogList): ActivityLogList.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: ActivityLogList, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): ActivityLogList;
  static deserializeBinaryFromReader(message: ActivityLogList, reader: jspb.BinaryReader): ActivityLogList;
}

export namespace ActivityLogList {
  export type AsObject = {
    resultsList: Array<ActivityLog.AsObject>,
    totalCount: number,
  }
}

