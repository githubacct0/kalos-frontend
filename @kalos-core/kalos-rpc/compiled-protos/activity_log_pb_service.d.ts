// package: 
// file: activity_log.proto

import * as activity_log_pb from "./activity_log_pb";
import {grpc} from "@improbable-eng/grpc-web";

type ActivityLogServiceCreate = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLog;
};

type ActivityLogServiceGet = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLog;
};

type ActivityLogServiceBatchGet = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLogList;
};

type ActivityLogServiceBatchGetEventLogs = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLogList;
};

type ActivityLogServiceList = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: true;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLog;
};

type ActivityLogServiceUpdate = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLog;
};

type ActivityLogServiceDelete = {
  readonly methodName: string;
  readonly service: typeof ActivityLogService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof activity_log_pb.ActivityLog;
  readonly responseType: typeof activity_log_pb.ActivityLog;
};

export class ActivityLogService {
  static readonly serviceName: string;
  static readonly Create: ActivityLogServiceCreate;
  static readonly Get: ActivityLogServiceGet;
  static readonly BatchGet: ActivityLogServiceBatchGet;
  static readonly BatchGetEventLogs: ActivityLogServiceBatchGetEventLogs;
  static readonly List: ActivityLogServiceList;
  static readonly Update: ActivityLogServiceUpdate;
  static readonly Delete: ActivityLogServiceDelete;
}

export type ServiceError = { message: string, code: number; metadata: grpc.Metadata }
export type Status = { details: string, code: number; metadata: grpc.Metadata }

interface UnaryResponse {
  cancel(): void;
}
interface ResponseStream<T> {
  cancel(): void;
  on(type: 'data', handler: (message: T) => void): ResponseStream<T>;
  on(type: 'end', handler: (status?: Status) => void): ResponseStream<T>;
  on(type: 'status', handler: (status: Status) => void): ResponseStream<T>;
}
interface RequestStream<T> {
  write(message: T): RequestStream<T>;
  end(): void;
  cancel(): void;
  on(type: 'end', handler: (status?: Status) => void): RequestStream<T>;
  on(type: 'status', handler: (status: Status) => void): RequestStream<T>;
}
interface BidirectionalStream<ReqT, ResT> {
  write(message: ReqT): BidirectionalStream<ReqT, ResT>;
  end(): void;
  cancel(): void;
  on(type: 'data', handler: (message: ResT) => void): BidirectionalStream<ReqT, ResT>;
  on(type: 'end', handler: (status?: Status) => void): BidirectionalStream<ReqT, ResT>;
  on(type: 'status', handler: (status: Status) => void): BidirectionalStream<ReqT, ResT>;
}

export class ActivityLogServiceClient {
  readonly serviceHost: string;

  constructor(serviceHost: string, options?: grpc.RpcOptions);
  create(
    requestMessage: activity_log_pb.ActivityLog,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  create(
    requestMessage: activity_log_pb.ActivityLog,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  get(
    requestMessage: activity_log_pb.ActivityLog,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  get(
    requestMessage: activity_log_pb.ActivityLog,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  batchGet(
    requestMessage: activity_log_pb.ActivityLog,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLogList|null) => void
  ): UnaryResponse;
  batchGet(
    requestMessage: activity_log_pb.ActivityLog,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLogList|null) => void
  ): UnaryResponse;
  batchGetEventLogs(
    requestMessage: activity_log_pb.ActivityLog,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLogList|null) => void
  ): UnaryResponse;
  batchGetEventLogs(
    requestMessage: activity_log_pb.ActivityLog,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLogList|null) => void
  ): UnaryResponse;
  list(requestMessage: activity_log_pb.ActivityLog, metadata?: grpc.Metadata): ResponseStream<activity_log_pb.ActivityLog>;
  update(
    requestMessage: activity_log_pb.ActivityLog,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  update(
    requestMessage: activity_log_pb.ActivityLog,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  delete(
    requestMessage: activity_log_pb.ActivityLog,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
  delete(
    requestMessage: activity_log_pb.ActivityLog,
    callback: (error: ServiceError|null, responseMessage: activity_log_pb.ActivityLog|null) => void
  ): UnaryResponse;
}

