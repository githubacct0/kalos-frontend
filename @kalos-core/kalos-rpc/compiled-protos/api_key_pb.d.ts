// package: 
// file: api_key.proto

import * as jspb from "google-protobuf";

export class ApiKey extends jspb.Message {
  getId(): number;
  setId(value: number): void;

  getTextId(): string;
  setTextId(value: string): void;

  getApiKey(): string;
  setApiKey(value: string): void;

  getApiEndpoint(): string;
  setApiEndpoint(value: string): void;

  getApiDescription(): string;
  setApiDescription(value: string): void;

  getApiUser(): number;
  setApiUser(value: number): void;

  clearFieldMaskList(): void;
  getFieldMaskList(): Array<string>;
  setFieldMaskList(value: Array<string>): void;
  addFieldMask(value: string, index?: number): string;

  getPageNumber(): number;
  setPageNumber(value: number): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): ApiKey.AsObject;
  static toObject(includeInstance: boolean, msg: ApiKey): ApiKey.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: ApiKey, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): ApiKey;
  static deserializeBinaryFromReader(message: ApiKey, reader: jspb.BinaryReader): ApiKey;
}

export namespace ApiKey {
  export type AsObject = {
    id: number,
    textId: string,
    apiKey: string,
    apiEndpoint: string,
    apiDescription: string,
    apiUser: number,
    fieldMaskList: Array<string>,
    pageNumber: number,
  }
}

export class ApiKeyList extends jspb.Message {
  clearResultsList(): void;
  getResultsList(): Array<ApiKey>;
  setResultsList(value: Array<ApiKey>): void;
  addResults(value?: ApiKey, index?: number): ApiKey;

  getTotalCount(): number;
  setTotalCount(value: number): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): ApiKeyList.AsObject;
  static toObject(includeInstance: boolean, msg: ApiKeyList): ApiKeyList.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: ApiKeyList, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): ApiKeyList;
  static deserializeBinaryFromReader(message: ApiKeyList, reader: jspb.BinaryReader): ApiKeyList;
}

export namespace ApiKeyList {
  export type AsObject = {
    resultsList: Array<ApiKey.AsObject>,
    totalCount: number,
  }
}

