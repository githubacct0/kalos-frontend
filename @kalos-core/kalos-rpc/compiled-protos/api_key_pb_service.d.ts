// package: 
// file: api_key.proto

import * as api_key_pb from "./api_key_pb";
import {grpc} from "@improbable-eng/grpc-web";

type ApiKeyServiceCreate = {
  readonly methodName: string;
  readonly service: typeof ApiKeyService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof api_key_pb.ApiKey;
  readonly responseType: typeof api_key_pb.ApiKey;
};

type ApiKeyServiceGet = {
  readonly methodName: string;
  readonly service: typeof ApiKeyService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof api_key_pb.ApiKey;
  readonly responseType: typeof api_key_pb.ApiKey;
};

type ApiKeyServiceBatchGet = {
  readonly methodName: string;
  readonly service: typeof ApiKeyService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof api_key_pb.ApiKey;
  readonly responseType: typeof api_key_pb.ApiKeyList;
};

type ApiKeyServiceList = {
  readonly methodName: string;
  readonly service: typeof ApiKeyService;
  readonly requestStream: false;
  readonly responseStream: true;
  readonly requestType: typeof api_key_pb.ApiKey;
  readonly responseType: typeof api_key_pb.ApiKey;
};

type ApiKeyServiceUpdate = {
  readonly methodName: string;
  readonly service: typeof ApiKeyService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof api_key_pb.ApiKey;
  readonly responseType: typeof api_key_pb.ApiKey;
};

type ApiKeyServiceDelete = {
  readonly methodName: string;
  readonly service: typeof ApiKeyService;
  readonly requestStream: false;
  readonly responseStream: false;
  readonly requestType: typeof api_key_pb.ApiKey;
  readonly responseType: typeof api_key_pb.ApiKey;
};

export class ApiKeyService {
  static readonly serviceName: string;
  static readonly Create: ApiKeyServiceCreate;
  static readonly Get: ApiKeyServiceGet;
  static readonly BatchGet: ApiKeyServiceBatchGet;
  static readonly List: ApiKeyServiceList;
  static readonly Update: ApiKeyServiceUpdate;
  static readonly Delete: ApiKeyServiceDelete;
}

export type ServiceError = { message: string, code: number; metadata: grpc.Metadata }
export type Status = { details: string, code: number; metadata: grpc.Metadata }

interface UnaryResponse {
  cancel(): void;
}
interface ResponseStream<T> {
  cancel(): void;
  on(type: 'data', handler: (message: T) => void): ResponseStream<T>;
  on(type: 'end', handler: (status?: Status) => void): ResponseStream<T>;
  on(type: 'status', handler: (status: Status) => void): ResponseStream<T>;
}
interface RequestStream<T> {
  write(message: T): RequestStream<T>;
  end(): void;
  cancel(): void;
  on(type: 'end', handler: (status?: Status) => void): RequestStream<T>;
  on(type: 'status', handler: (status: Status) => void): RequestStream<T>;
}
interface BidirectionalStream<ReqT, ResT> {
  write(message: ReqT): BidirectionalStream<ReqT, ResT>;
  end(): void;
  cancel(): void;
  on(type: 'data', handler: (message: ResT) => void): BidirectionalStream<ReqT, ResT>;
  on(type: 'end', handler: (status?: Status) => void): BidirectionalStream<ReqT, ResT>;
  on(type: 'status', handler: (status: Status) => void): BidirectionalStream<ReqT, ResT>;
}

export class ApiKeyServiceClient {
  readonly serviceHost: string;

  constructor(serviceHost: string, options?: grpc.RpcOptions);
  create(
    requestMessage: api_key_pb.ApiKey,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  create(
    requestMessage: api_key_pb.ApiKey,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  get(
    requestMessage: api_key_pb.ApiKey,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  get(
    requestMessage: api_key_pb.ApiKey,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  batchGet(
    requestMessage: api_key_pb.ApiKey,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKeyList|null) => void
  ): UnaryResponse;
  batchGet(
    requestMessage: api_key_pb.ApiKey,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKeyList|null) => void
  ): UnaryResponse;
  list(requestMessage: api_key_pb.ApiKey, metadata?: grpc.Metadata): ResponseStream<api_key_pb.ApiKey>;
  update(
    requestMessage: api_key_pb.ApiKey,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  update(
    requestMessage: api_key_pb.ApiKey,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  delete(
    requestMessage: api_key_pb.ApiKey,
    metadata: grpc.Metadata,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
  delete(
    requestMessage: api_key_pb.ApiKey,
    callback: (error: ServiceError|null, responseMessage: api_key_pb.ApiKey|null) => void
  ): UnaryResponse;
}

