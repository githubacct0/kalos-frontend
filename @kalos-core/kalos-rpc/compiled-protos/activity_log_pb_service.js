// package: 
// file: activity_log.proto

var activity_log_pb = require("./activity_log_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var ActivityLogService = (function () {
  function ActivityLogService() {}
  ActivityLogService.serviceName = "ActivityLogService";
  return ActivityLogService;
}());

ActivityLogService.Create = {
  methodName: "Create",
  service: ActivityLogService,
  requestStream: false,
  responseStream: false,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLog
};

ActivityLogService.Get = {
  methodName: "Get",
  service: ActivityLogService,
  requestStream: false,
  responseStream: false,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLog
};

ActivityLogService.BatchGet = {
  methodName: "BatchGet",
  service: ActivityLogService,
  requestStream: false,
  responseStream: false,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLogList
};

ActivityLogService.BatchGetEventLogs = {
  methodName: "BatchGetEventLogs",
  service: ActivityLogService,
  requestStream: false,
  responseStream: false,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLogList
};

ActivityLogService.List = {
  methodName: "List",
  service: ActivityLogService,
  requestStream: false,
  responseStream: true,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLog
};

ActivityLogService.Update = {
  methodName: "Update",
  service: ActivityLogService,
  requestStream: false,
  responseStream: false,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLog
};

ActivityLogService.Delete = {
  methodName: "Delete",
  service: ActivityLogService,
  requestStream: false,
  responseStream: false,
  requestType: activity_log_pb.ActivityLog,
  responseType: activity_log_pb.ActivityLog
};

exports.ActivityLogService = ActivityLogService;

function ActivityLogServiceClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

ActivityLogServiceClient.prototype.create = function create(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(ActivityLogService.Create, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

ActivityLogServiceClient.prototype.get = function get(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(ActivityLogService.Get, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

ActivityLogServiceClient.prototype.batchGet = function batchGet(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(ActivityLogService.BatchGet, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

ActivityLogServiceClient.prototype.batchGetEventLogs = function batchGetEventLogs(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(ActivityLogService.BatchGetEventLogs, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

ActivityLogServiceClient.prototype.list = function list(requestMessage, metadata) {
  var listeners = {
    data: [],
    end: [],
    status: []
  };
  var client = grpc.invoke(ActivityLogService.List, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onMessage: function (responseMessage) {
      listeners.data.forEach(function (handler) {
        handler(responseMessage);
      });
    },
    onEnd: function (status, statusMessage, trailers) {
      listeners.status.forEach(function (handler) {
        handler({ code: status, details: statusMessage, metadata: trailers });
      });
      listeners.end.forEach(function (handler) {
        handler({ code: status, details: statusMessage, metadata: trailers });
      });
      listeners = null;
    }
  });
  return {
    on: function (type, handler) {
      listeners[type].push(handler);
      return this;
    },
    cancel: function () {
      listeners = null;
      client.close();
    }
  };
};

ActivityLogServiceClient.prototype.update = function update(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(ActivityLogService.Update, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

ActivityLogServiceClient.prototype.delete = function pb_delete(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(ActivityLogService.Delete, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.ActivityLogServiceClient = ActivityLogServiceClient;

