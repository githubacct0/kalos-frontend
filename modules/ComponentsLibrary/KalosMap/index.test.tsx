import React from 'react';
import renderer from 'react-test-renderer';
import { KalosMap } from './index';
import { Place } from '@kalos-core/kalos-rpc/compiled-protos/kalosmaps_pb';

test('SearchIndex renders correctly', () => {
  const origin = new Place(),
    destination = new Place();

  origin.setStreetNumber(3030);
  origin.setRoadName('US Hwy 27 S');
  origin.setCity('Sebring');
  origin.setState('FL');
  origin.setCountry('United States');

  destination.setStreetNumber(75);
  destination.setRoadName('Harold St');
  destination.setCity('Sydney');
  destination.setState('NS B1P 3M2');
  destination.setCountry('Canada');

  let distance = '';

  const tree = renderer
    .create(
      <KalosMap
        isOpen
        Addresses={{
          origin: [origin],
          destination: destination,
        }}
        onDistanceCalculated={(dist: any) => {
          distance = dist;
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot(); // Rendered correctly
});
