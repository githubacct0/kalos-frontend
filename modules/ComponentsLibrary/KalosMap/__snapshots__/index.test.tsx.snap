// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchIndex renders correctly 1`] = `null`;
